src/jdk.charsets/share/classes/sun/nio/cs/ext/AbstractCharsetProvider.java

```diff
@@ -68,11 +68,11 @@
     private String packagePrefix;
 
     protected AbstractCharsetProvider() {
-        packagePrefix = "sun.nio.cs";
+        packagePrefix = "sun.nio.cs.";
     }
 
     protected AbstractCharsetProvider(String pkgPrefixName) {
-        packagePrefix = pkgPrefixName;
+        packagePrefix = pkgPrefixName.concat(".");
```

Comment:
Hm, I wonder why not just `pkgPrefixName + '.'` here and below?

Reply:
Yes, I wanted to measure the overhead of `Charset` class initialization done by `Charset.availableCharsets()` and the `StringConcatFactory` bootstraps was a reasonable chunk of the cost so I moved them out of the picture. I didn't mind what I ended up with, but if you prefer I can move back to `pkgPrefixName + '.'` here.

Reply:
Let's keep it as is, to me it's a readable as concatenation via `+`